### PR TITLE
Implement anonymous traversal

### PR DESCRIPTION
**Pull request by simonz-bq**
*Issue #, if available:*

https://bitquill.atlassian.net/browse/AN-1036

*Description of changes:*

Polish anonymous traversal implementation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gremlin-go/driver/anonymousTraversal.go [hunk 1]
```diff
@@ -19,534 +19,766 @@ under the License.
 
 package gremlingo
```
Comment (lyndonbauto) [inline gremlin-go/driver/anonymousTraversal.go:20]: Did you intentionally leave out the '.' for all your comments?

File: gremlin-go/driver/connection_test.go [hunk 2]
```diff
@@ -130,6 +130,15 @@ func readCount(t *testing.T, g *GraphTraversalSource, label string, expected int
 	assert.Equal(t, int32(expected), count)
 }
 
+func readUsingAnonymousTraversal(t *testing.T, g *GraphTraversalSource) {
+	results, err := g.V().Fold().Project(testLabel, personLabel).By(T__.Unfold().HasLabel(testLabel).Count()).By(T__.Unfold().HasLabel(personLabel).Count()).ToList()
```
Comment (lyndonbauto) [inline gremlin-go/driver/connection_test.go]: For readability, I suggest:
```suggestion
	results, err := g.V().Fold().
		Project(testLabel, personLabel).
			By(T__.Unfold().HasLabel(testLabel).Count()).
			By(T__.Unfold().HasLabel(personLabel).Count()).
		ToList()
```

File: gremlin-go/driver/anonymousTraversal.go (hunk 1 above)
Comment (simonz-bq) [reply gremlin-go/driver/anonymousTraversal.go:20]: Are they supposed to be there? They originally didn't exist on the AnonymousTraversal

Comment (lyndonbauto) [reply gremlin-go/driver/anonymousTraversal.go:20]: Not sure, personally IDC, I am pretty sure it does not affect the quality of the comment in any measurable way.

**Review by lyndonbauto — APPROVED** [review]
:shipit:

**Pull request merged**